### PR TITLE
Add Reducers and Actions

### PR DESCRIPTION
**Pull request by HansZizold**
** CHANGES **
- Redux Toolkit added.
- Folder structure changed as suggested.
- Actions and reducers were implemented.
- Both reducers were combined into a root Reducer.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const ADDBOOK = 'bookstore/src/redux/books/ADD_BOOK';
+const REMOVEBOOK = 'bookstore/src/redux/books/REMOVE_BOOK';
+
```
Comment (zieeco) [inline src/redux/books/books.js]: - Please, kindly restructure this to follow the project requirement and conform to using the right duck's pattern below:
``` 
const ADDBOOK = 'bookstore/books/ADD_BOOK';
const REMOVEBOOK = 'bookstore/src/books/REMOVE_BOOK';
```

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  title: 'Book 0',
+  author: 'Author 0',
+};
+
+export const removeBook = {
+  type: REMOVEBOOK,
+  id: 0,
+};
+
+const bookReducer = (state = [], action) => {
```
Comment (zieeco) [inline src/redux/books/books.js]: - Please, kindly create an initialState variable as required in the project requirement like this

` const initialState = [];`
`const bookReducer = (state = initialState, action) => {`

File: src/redux/categories/categories.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const CHECKCATEGORIES = 'bookstore/src/redux/categories/CHECK_CATEGORIES';
```
Comment (zieeco) [inline src/redux/categories/categories.js]: - Please, kindly restructure this to follow the project requirement and conform to using the right duck's pattern below:
``` 
const CHECKCATEGORIES = 'bookstore/categories/CHECK_CATEGORIES';

```

File: src/redux/categories/categories.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+export const checkcategories = {
+  type: CHECKCATEGORIES,
+  id: 0,
+};
+
+const checkReducer = (state = [], action) => {
```
Comment (zieeco) [inline src/redux/categories/categories.js]: - Please, kindly create an initialState variable as required in the project requirement like this

` const initialState = [];`
`const checkReducer = (state = initialState, action) => {`

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @HansZizold,


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (HansZizold) [reply src/redux/books/books.js]: @zieeco, I think I misunderstood the meaning of this. But I will change as you are indicating to me.
To be honest I am still a little confused.

File: src/redux/books/books.js (hunk 2 above)
Comment (HansZizold) [reply src/redux/books/books.js]: Oki, it is done

File: src/redux/categories/categories.js (hunk 3 above)
Comment (HansZizold) [reply src/redux/categories/categories.js]: Oki, it is done!

File: src/redux/categories/categories.js (hunk 4 above)
Comment (HansZizold) [reply src/redux/categories/categories.js]: Oki, it is done!

**Comment by HansZizold** [comment]
Hi @zieeco, thank you very much for your revision.
I think I remediated the observations. Please check again.

File: src/redux/books/books.js (hunk 1 above)
Comment (TedLivist) [reply src/redux/books/books.js]: Hi @HansZizold you can refer to [this article](https://medium.com/@matthew.holman/what-is-redux-ducks-46bcb1ad04b7) for better understanding of the ducks pattern or [this repo demostrating it](https://github.com/erikras/ducks-modular-redux).

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @HansZizold,

✔️ Implementation of redux ducks
✔️ Descriptive PR Summary

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**